### PR TITLE
[Relayminer, Bug] fix: sessiontree logger never initialized

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Fixes a nil pointer panic in the relayminer; the logger member is never initialized.

## Issue

- N/A

## Type of change

Select one or more from the following:

- [ ] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [x] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [ ] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

**Comment by bryanchriswhite** [comment]
@red-0ne can you confirm whether this is a duplicate of #1007 and if there's anything that doesn't overlap that we should keep (e.g. perhaps moving the `.With()` up in scope).

**Comment by red-0ne** [comment]
@bryanchriswhite , Yes it's a duplicate with some minor differences.

The logger's `With` is something we should keep :+1:

File: pkg/relayer/session/sessiontree.go [hunk 1]
```diff
@@ -95,7 +96,13 @@ func NewSessionTree(
 	// contain a non-hashed Relay that could be used to validate the proof on-chain.
 	trie := smt.NewSparseMerkleSumTrie(treeStore, protocol.NewTrieHasher(), smt.WithValueHasher(nil))
 
+	logger := polylog.Ctx(context.TODO()).With(
```
Comment (red-0ne) [inline pkg/relayer/session/sessiontree.go:100]: `logger` should be passed to `NewSessionTree`.

We should keep the `With` method though.

File: pkg/relayer/session/sessiontree.go [hunk 2]
```diff
@@ -96,7 +97,13 @@ func NewSessionTree(
...
+		"session_id", sessionHeader.SessionId,
+		"supplier_operator_address", supplierOperatorAddress,
+	)
+
 	sessionTree := &sessionTree{
+		logger:                  logger,
```
Comment (red-0ne) [inline pkg/relayer/session/sessiontree.go:106]: `logger` keys is already present, we should keep only one.

**Review by red-0ne — APPROVED** [review]

**Pull request merged**